### PR TITLE
Incorporate Eloquent Methods into Model Generation

### PR DESCRIPTION
**Pull request by jschlies**
Incorporate Eloquent Methods into Model Generation 
- Add code to query default DB and using schema info, generate Eloquent methods
- created templates/models/
- Added @version phpdoc tag using a timestamp

I'm open to suggestions.

### CODE REVIEW

File: src/Generators/ModelGenerator.php [hunk 1]
```diff
@@ -18,16 +19,55 @@ class ModelGenerator extends BaseGenerator
...
+        $this->getSchemaManager();
+        $this->getTables();
+        $this->getColumnsPrimaryAndForeignKeysPerTable();
+        $this->getEloquentRules();
     }
 
     public function generate()
     {
-        $templateData = TemplateUtil::getTemplate('model', 'laravel-generator');
+        $templateData = TemplateUtil::getTemplate('models.model', 'laravel-generator');
```
Comment (jschlies) [inline src/Generators/ModelGenerator.php]: Unsure how you want to handle this as users who use published will need to react to new .stub file scructure

File: src/Generators/ModelGenerator.php [hunk 2]
```diff
@@ -235,4 +282,441 @@ public function rollback()
             $this->commandData->commandComment('Model file deleted: '.$this->fileName);
         }
     }
+
+    private function getEloquentRules()
```
Comment (jschlies) [inline src/Generators/ModelGenerator.php]: Thought about creating a stand alone generateEloquent class. Perhaps in next rev.

File: templates/models/model.stub [hunk 3]
```diff
@@ -33,4 +33,6 @@ $PRIMARY$
     public static $rules = [
         $RULES$
```
Comment (jschlies) [inline templates/models/model.stub]: Note the file location change

File: templates/models/hasMany.stub [hunk 4]
```diff
@@ -0,0 +1,11 @@
+/**
```
Comment (jschlies) [inline templates/models/hasMany.stub]: in next rev, I want to add functions like:
public function attachModel($model)
public function attachModels($model)
public function detachModel($model)
public function detachModels($model)

**Comment by jschlies** [comment]
Mitul,

I just merged resent changes to InfyOmLabs/laravel-generator:master. Tested and everything I'm using works

**Comment by mitulgolakiya** [comment]
@jschlies, Thanks. I will review everything by this weekend and will merge it.

**Comment by jschlies** [comment]
@mitulgolakiya - Is there something I can do or any information I can offer to assist?? I'd be happy to update/revise my changes if there are concerns.

**Comment by mitulgolakiya** [comment]
Thanks @jschlies, Actually I am bit busy in some stuff so missed it. But I will merge it this weekend and if anything is needed then will let you know.

**Comment by jschlies** [comment]
Same message as above I guess. Is there something I can do or any information I can offer to assist?? I'd be happy to update/revise my changes if there are concerns.

**Comment by mitulgolakiya** [comment]
@jschlies Can you close this PR and put a new one with develop branch ? so I can do testing after merging it.
Actually, now we have develop branch we don't want to break master branch as people are using that with their production stuff may be.

And sorry for late reply.

**Comment by jschlies** [comment]
No problem. Will do by Monday US time.

Jim

**Pull request closed without merging**